tests/db_access/connection_pool.cpp

```diff
@@ -34,67 +33,55 @@ go_bandit([](){
     describe("Test connection pool", [&](){
         auto& s = getSettingsRef();
 
+
         it("can clear pool properly", [&](){
             auto&& connectionPool = makeConnectionPool<true, true, true, true, true>([&](){
                 return std::async(std::launch::async, [&](){ return std::make_shared<Connection>(s.database, s.user, s.password, s.host, s.port); });
             });
+            auto waitForPoolState = [&connectionPool](std::chrono::milliseconds max, auto size, auto available) {
```

Comment:
In this case, I'd prefer `std::size_t` over `auto`.

Reply:
Also, this implementation of these conveniency functions is totally ok, but considering the fact that you use them in multiple tests, it may be better overall to define them as conventional, standalone static functions with explicit connectionPool parameter.